parquet-variant/src/encoder/variant_to_json.rs

```diff
@@ -0,0 +1,993 @@
+
+        let mut builder = VariantBuilder::new();
+
+        {
+            let mut list = builder.new_list();
+            list.append_value("string_value");
+            list.append_value(42i32);
+            list.append_value(true);
+            list.append_value(std::f64::consts::PI);
+            list.append_value(false);
+            list.append_value(()); // null
+            list.append_value(100i64);
+            list.finish();
+        }
+
+        let (metadata, value) = builder.finish();
+        let variant = Variant::try_new(&metadata, &value)?;
+        let json = variant_to_json_string(&variant)?;
+
+        let parsed: Value = serde_json::from_str(&json)
+            .map_err(|e| ArrowError::ParseError(format!("JSON parse error: {}", e)))?;
+        if let Value::Array(arr) = parsed {
+            assert_eq!(arr.len(), 7);
+            assert_eq!(arr[0], Value::String("string_value".to_string()));
+            assert_eq!(arr[1], Value::Number(42.into()));
+            assert_eq!(arr[2], Value::Bool(true));
+            assert!(matches!(arr[3], Value::Number(_))); // float
+            assert_eq!(arr[4], Value::Bool(false));
+            assert_eq!(arr[5], Value::Null);
+            assert_eq!(arr[6], Value::Number(100.into()));
+        } else {
+            panic!("Expected JSON array");
+        }
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_object_with_various_primitive_types_to_json() -> Result<(), ArrowError> {
+        use crate::builder::VariantBuilder;
+
+        let mut builder = VariantBuilder::new();
+
+        {
+            let mut obj = builder.new_object();
+            obj.append_value("string_field", "test_string");
+            obj.append_value("int_field", 123i32);
+            obj.append_value("bool_field", true);
+            obj.append_value("float_field", 2.71f64);
+            obj.append_value("null_field", ());
+            obj.append_value("long_field", 999i64);
+            obj.finish();
+        }
+
+        let (metadata, value) = builder.finish();
+        let variant = Variant::try_new(&metadata, &value)?;
+        let json = variant_to_json_string(&variant)?;
+
+        let parsed: Value = serde_json::from_str(&json)
+            .map_err(|e| ArrowError::ParseError(format!("JSON parse error: {}", e)))?;
+        if let Value::Object(obj) = parsed {
+            assert_eq!(obj.len(), 6);
+            assert_eq!(
+                obj.get("string_field"),
+                Some(&Value::String("test_string".to_string()))
+            );
+            assert_eq!(obj.get("int_field"), Some(&Value::Number(123.into())));
+            assert_eq!(obj.get("bool_field"), Some(&Value::Bool(true)));
+            assert!(matches!(obj.get("float_field"), Some(Value::Number(_))));
+            assert_eq!(obj.get("null_field"), Some(&Value::Null));
+            assert_eq!(obj.get("long_field"), Some(&Value::Number(999.into())));
+        } else {
+            panic!("Expected JSON object");
+        }
```

Comment:
FWIW if you like less intending you could use this pattern too (just pointing this out)

```suggestion
        let Value::Object(obj) = parsed else {
            panic!("Expected JSON object");
        }
        assert_eq!(obj.len(), 6);
        assert_eq!(
            obj.get("string_field"),
            Some(&Value::String("test_string".to_string()))
        );
        assert_eq!(obj.get("int_field"), Some(&Value::Number(123.into())));
        assert_eq!(obj.get("bool_field"), Some(&Value::Bool(true)));
        assert!(matches!(obj.get("float_field"), Some(Value::Number(_))));
        assert_eq!(obj.get("null_field"), Some(&Value::Null));
        assert_eq!(obj.get("long_field"), Some(&Value::Number(999.into())));
```